### PR TITLE
[doc] fix issue #7260 by refactoring main "Getting Started" page

### LINKED ISSUE #7260: [doc] too many "Getting Started" pages make its difficult to follow instructions
In order to make the project more appealing to newcomers I propose:
1. Moving the master [Getting Started](https://docs.opentitan.org/doc/ug/getting_started/) page to the [top level documentation page](https://docs.opentitan.org/).
2. Refactoring the [Getting Started](https://docs.opentitan.org/doc/ug/getting_started/) page to: a) make it more concise, b) add a flowchart highlighting instructions, c) incorporate/promote use of Docker container with correct build environment to get started faster with (if verilator is your only target).

### PR DESCRIPTION
**Pull request by timothytrippel**
Fix stream-lines main "Getting Started" page by:
1. Adding a figure to show different workflows.
2. Detailing the exact steps needed to get a "Hello World!" demo up and running on a target of choice.
3. Provides pointers to additional resources about individual aspects (general, HW, SW) of the OpenTitan project.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

**Comment by timothytrippel** [comment]
@tjaychen @rswarbrick @mundaym @alphan @imphil 
please review

File: doc/ug/getting_started/index.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+There are a set of [notes for using GitHub]({{< relref "github_notes.md" >}}) which explains how to work with your own fork.
+
+***Note: throughout the documentation `$REPO_TOP` refers to the path where the OpenTitan repository is checked out.***
+
+## Building OpenTitan Software/Hardware
+
+Once you have setup a suitable environment and cloned the OpenTitan source code, you need to choose a top-level OpenTitan design to build.
+The code base contains multiple top-level designs, listed [here]({{< relref "/doc/ug/system_list.md" >}}), which can be synthesized or compiled for various targets.
+A target can be a simulation tool, an FPGA board, or an ASIC technology.
+We currently support two out of the three aformentioned target types (i.e., simulation tools and FPGA), as described in this table:
```
Comment (imphil) [inline doc/ug/getting_started/index.md:41]: I know the wording we have in many places is blurry, but let's at least clarify things slightly here what "Verilator" vs "DV" means:

* What is called here "Verilator" is a cycle-accurate chip simulation primarily aimed at software development. (Yes, it's produced by Verilator, but that's mostly an implementation detail.)
* What is called "DV" is the hardware verification environment, primarily a constraint-random verification environment for individual hardware blocks. It also provide a full-chip simulation, but we use that mostly for testing the hardware in combination with some software, not developing software itself. The "DV" simulation can be powered by different simulators, VCS and Xcelium being the two primary ones.

**Review by imphil — APPROVED** [review]
That's a nice improvement, thanks! Some comments inline.

File: doc/ug/getting_started/index.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+This guide will help you get OpenTitan up and running by instructing you how to:
+
+1. setup an adequate build/testing environment on your machine,
+2. clone the OpenTitan Git repository, and
+3. build OpenTitan software/hardware for the target of your choosing.
+
+![Getting Started Workflow](getting_started_workflow.svg)
+
+As shown in the diagram above, we currently support multiple build targets and workflows, including: Verilator, FPGA, and DV.
+However, if you are new to the project, we recommend trying the Verilator workflow, as this uses only free EDA tools, and does not require any additional hardware.
```
[the remark this answers is not in the document]
Comment (rswarbrick) [reply doc/ug/getting_started/index.md:16]: Maybe "... we recommend simulation with Verilator ..." ?

File: _index.md [hunk 3]
```diff
@@ -7,12 +7,12 @@ OpenTitan will make the silicon RoT design and implementation more transparent,
...
 
-To get started using or contributing to the OpenTitan codebase, see the
-[list of user guides]({{< relref "doc/ug" >}}).
-For details on coding styles or how to use our project-specific tooling, see the
-[reference manuals]({{< relref "doc/rm" >}}).
-[This page]({{< relref "hw" >}})
-contains technical documentation on the SoC, the Ibex processor core, and the individual IP blocks.
+## Getting Started
+
+To get started _using_ OpenTitan, see the [Getting Started]({{< relref "doc/ug/getting_started" >}}) page.
```
[the remark this answers is not in the document]
Comment (timothytrippel) [reply _index.md:12]: I was thinking a *user* may be an academic researcher looking to *use* OpenTitan as a surrogate design for testing various hardware-related technologies (e.g. my hardware fuzzing work). Such a party/person would probably not be as interested in the various style guides and/or hw/sw design guides aimed at contributors or those modifying the design, rather just a consumer of the design. However, after reflection I think this distinction will probably only cause confusion, so I will reword as suggested :).

File: doc/ug/getting_started/index.md (hunk 1 above)
Comment (timothytrippel) [reply doc/ug/getting_started/index.md:41]: @imphil I updated the table to indicate the primary purpose of each target. Let me know if you think this is OK? or if I should remove it?

Comment (imphil) [reply doc/ug/getting_started/index.md:41]: Thanks for the update, looks good!

**Comment by imphil** [comment]
Thanks @timothytrippel, LGTM! I'll leave this open over the weekend to give others a chance to jump in and will merge on Monday if nothing else comes up.

**Review by alphan — APPROVED** [review]
Thanks @timothytrippel!

File: doc/ug/getting_started/index.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+## Building OpenTitan Software/Hardware
+
+Once you have setup a suitable environment and cloned the OpenTitan source code, you need to choose a top-level OpenTitan design to build.
+The code base contains multiple [top-level designs]({{< relref "/doc/ug/system_list.md" >}}), which can be synthesized or simulated.
+A target can be a simulation tool, an FPGA board, or an ASIC technology.
+We currently support two out of the three aformentioned target types (i.e., simulation tools and FPGA), as described in this table:
+| Target| Type | Primary Purpose | EDA Licenses Required? | Extra Hardware Required? |
+|-------|------|-----------------|------------------------|--------------------------|
+| [Verilator]({{< relref "getting_started_verilator.md" >}})| Simulation Tool | SW Testing | no | no |
```
Comment (mundaym) [inline doc/ug/getting_started/index.md:44]: It took me a while to realise that I was supposed to click on these links when looking at the rendered page. Maybe we could add a link column that mentioned the page name explicitly:

```
| Link |
|-------|
| [Getting Started with Verilator](...) |
```

File: doc/ug/getting_started/index.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+## Environment Setup
+
+There are two choices for setting up an OpenTitan environment:
+1. \[Supported\] [manual installation]({{< relref "install_instructions" >}}), or
+2. \[Experimental\] [provided Docker container]({{< relref "/util/container/README.md" >}}).
+
+While the first option is the only one currently supported, the second option is the quickest to get running.
+If you plan on being a long term user of the project, and/or contributer, we recommend the first (manual installation) option.
+Alternatively, if your goal is to get a Verilator simulation up and running as quickly as possible, we recommend the second (container) option.
```
Comment (mundaym) [inline doc/ug/getting_started/index.md:26]: Maybe `s/we recommend/you may find it easier to try/` or similar? I'm not sure we should recommend it that directly yet.

**Review by mundaym — APPROVED** [review]
Thanks @timothytrippel, looks like a good improvement to me!

**Pull request merged**